src/target/riscv/riscv-013.c

```diff
@@ -2892,17 +2891,27 @@ static int riscv013_test_sba_config_reg(struct target *target,
 		LOG_INFO("System Bus Access Test 1: Read/write test, no addr autoincrement PASSED");
 
 	// Test 2: Simple write/read test, with address autoincrement 
+	uint32_t curr_addr;
+	uint32_t prev_addr;
 	test_passed = true;
 	sbcs = set_field(sbcs_orig, DMI_SBCS_SBAUTOINCREMENT, 1);
 	dmi_write(target, DMI_SBCS, sbcs);
 
-	for (int sbaccess = 0; sbaccess <= max_sbaccess; sbaccess++) {
+	for (uint32_t sbaccess = 0; sbaccess <= (uint32_t)max_sbaccess; sbaccess++) {
```

Comment:
Why did you change this type to uint32_t? It looks like it just got you a lot of trouble.